### PR TITLE
(FM-7922) running the configuration to apply catalog to transport 

### PR DESCRIPTION
**Pull request by Thomas-Franklin**
The commits here are just for verification and will be removed before this is considered ready.

### CODE REVIEW

**Comment by Thomas-Franklin** [comment]
Some notes on verifying; make sure the puppetserver and puppetdb containers are running in spec folder. Create a catalog, i.e.:

Site.pp:

```
node 'vvtzckq3vzx995w.delivery.puppetlabs.net' {
panos_tag {
  'minimal':
    ensure => 'present';
  'Test Tag':
    ensure   => 'present',
    color    => 'red',
    comments => 'comments 123';
}
}
```

Copy it over to container: `docker cp ~/site.pp spec_puppet_1:/etc/puppetlabs/code/environments/production/manifests/site.pp`

Run the `'acceptance'` test: `bundle exec rspec spec/acceptance/ace/transport_app_spec.rb`

Verify the report is uploaded:

```
curl -X GET https://0.0.0.0:32776/pdb/query/v4 --tlsv1  --cacert ./spec/volumes/puppet/ssl/certs/ca.pem --cert ./spec/volumes/puppet/ssl/certs/aceserver.pem  --key ./spec/volumes/puppet/ssl/private_keys/aceserver.pem --data-urlencode 'query=reports {
  latest_report? = true
}' -k
```

File: spec/acceptance/ace/transport_app_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        "remote-transport": "panos",
+        host: certname,
+        user: "admin",
+        password: "admin",
+        ssl: false
+      },
+      "compiler": {
+        "certname": certname,
+        "environment": "production",
+        "transaction_uuid": Random.new_seed.to_s,
```
Comment (da-ar) [inline spec/acceptance/ace/transport_app_spec.rb]: manual verification finds that this is throwing an error on the reporting side:

`2019-04-12 14:47:28,622 ERROR [p.p.command] [5] [store report] Retrying after attempt 0 for livupbi6o63eyg3.delivery.puppetlabs.net, due to: org.postgresql.util.PSQLException: ERROR: invalid input syntax for uuid: "255097297177477792347640402086542686387"`

The suggestion is to replace with a hard-coded UUID value e.g. "2d931510-d99f-494a-8c67-87feb05e1594"

File: spec/acceptance/ace/transport_app_spec.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        "job_id": Random.new_seed.to_s
```
Comment (da-ar) [inline spec/acceptance/ace/transport_app_spec.rb]: following on from above, hard-code this value to something known e.g. 555

**Comment by da-ar** [comment]
Found that the port value in this command can change. We may want to revisit the Dockerfile to set this to a fix value.

> curl -X GET https://0.0.0.0:32776/pdb/query/v4 --tlsv1  --cacert ./spec/volumes/puppet/ssl/certs/ca.pem --cert ./spec/volumes/puppet/ssl/certs/aceserver.pem  --key ./spec/volumes/puppet/ssl/private_keys/aceserver.pem --data-urlencode 'query=reports {
>   latest_report? = true
> }' -k

File: spec/acceptance/ace/transport_app_spec.rb (hunk 1 above)
Comment (Thomas-Franklin) [reply spec/acceptance/ace/transport_app_spec.rb]: Thanks for finding it, not sure why this wasn't shown, was this trailing the PuppetDB logs? The acceptance test is going be thrown away afterwards, was just to have it contained somewhere to verify the refactor etc.

Comment (Thomas-Franklin) [reply spec/acceptance/ace/transport_app_spec.rb]: Ah yes, I see it now.

**Comment by Thomas-Franklin** [comment]
> Found that the port value in this command can change. We may want to revisit the Dockerfile to set this to a fix value.
> 
> > curl -X GET https://0.0.0.0:32776/pdb/query/v4 --tlsv1  --cacert ./spec/volumes/puppet/ssl/certs/ca.pem --cert ./spec/volumes/puppet/ssl/certs/aceserver.pem  --key ./spec/volumes/puppet/ssl/private_keys/aceserver.pem --data-urlencode 'query=reports {
> > latest_report? = true
> > }' -k

Not something we will be doing often, but yea we can update the compose file to assign it a fixed port

**Pull request merged**